api/core/tools/provider/builtin/databaseoperation/databaseoperation.py

```diff
@@ -0,0 +1,10 @@
+from core.tools.errors import ToolProviderCredentialValidationError
+from core.tools.provider.builtin_tool_provider import BuiltinToolProviderController
+
+
+class DatabaseOperationProvider(BuiltinToolProviderController):
+    def _validate_credentials(self, credentials: dict) -> None:
+        try:
+            pass
```

Comment:
At least init a tool instance with no args anyway, making sure the tool class is loadable?